internal/testrunner/runners/pipeline/test_result.go

```diff
@@ -0,0 +1,84 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/kylelemons/godebug/diff"
+	"github.com/pkg/errors"
+)
+
+const expectedTestResultSuffix = "-expected.json"
+
+var errTestCaseFailed = errors.New("test case failed")
+
+type testResult struct {
+	events []json.RawMessage
+}
+
+type testResultDefinition struct {
+	Expected []json.RawMessage `json:"expected"`
+}
+
+func writeTestResult(testCasePath string, result *testResult) error {
+	testCaseDir := filepath.Dir(testCasePath)
+	testCaseFile := filepath.Base(testCasePath)
+
+	data, err := marshalTestResult(result)
+	if err != nil {
+		return errors.Wrap(err, "marshalling test result failed")
+	}
+	err = ioutil.WriteFile(filepath.Join(testCaseDir, expectedTestResultFile(testCaseFile)), data, 0644)
+	if err != nil {
+		return errors.Wrap(err, "writing test result failed")
+	}
+	return nil
+}
+
+func compareResults(testCasePath string, result *testResult) error {
+	current, err := marshalTestResult(result)
+	if err != nil {
+		return errors.Wrap(err, "marshalling test result failed")
+	}
+
+	expected, err := readExpectedTestResult(testCasePath)
+	if err != nil {
+		return errors.Wrap(err, "reading expected test result failed")
+	}
+
+	report := diff.Diff(string(expected), string(current))
+	if report != "" {
+		fmt.Println("Expected results are different from current ones:")
+		fmt.Println(report)
```

Comment:
Should the first or both statements here use the logger?

Reply:
Based on our discussion in https://github.com/elastic/elastic-package/pull/73/files#r475905214, I would consider this to be part of the output of the command, not about what the command is doing during it's execution. So I'm good with what you have here.

Reply:
Fixed

Reply:
@mtojek I think you misread my comment 🙂. I was good with what you had here before (`fmt.Println`).

Reply:
I apologize for the confusion. It's fixed back :)